packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5621,10 +5621,10 @@ class _HistoryProperty extends RestorableProperty<Map<String?, List<Object>>?> {
 
   @override
   Map<String?, List<Object>>? fromPrimitives(Object? data) {
-    final Map<dynamic, dynamic> casted = data! as Map<dynamic, dynamic>;
-    return casted.map<String?, List<Object>>((dynamic key, dynamic value) => MapEntry<String?, List<Object>>(
+    final Map<Object?, Object?> casted = data! as Map<Object?, Object?>;
+    return casted.map<String?, List<Object>>((Object? key, Object? value) => MapEntry<String?, List<Object>>(
       key as String?,
-      List<Object>.from(value as List<dynamic>, growable: true),
+      List<Object>.from(value as List<Object?>, growable: true),
```

Comment:
This is causing the analyzer failure. I believe it's expecting a `List<Object>`, but you're giving it a `List<Object?>`